app/models/miq_report/seeding.rb

```diff
@@ -84,12 +86,34 @@ def seed_record(path, report)
       end
     end
 
+    def template_type(path)
+      path.include?(RELATIVE_REPORT_DIR.to_s) ? "report" : "compare"
+    end
+
     def seed_files
+      seed_core_files + seed_plugin_files
+    end
+
+    def seed_core_files
       Dir.glob(REPORT_DIR.join("**/*.yaml")).sort + Dir.glob(COMPARE_DIR.join("**/*.yaml")).sort
     end
 
+    def seed_plugin_files
+      Vmdb::Plugins.flat_map do |plugin|
+        Dir.glob(plugin.root.join("#{RELATIVE_REPORT_DIR}/**/*.yaml")).sort + Dir.glob(plugin.root.join("#{RELATIVE_COMPARE_DIR}/**/*.yaml")).sort
```

Comment:
`seed_core_files` and `seed_plugin_files` seem pretty similar.  Wonder if you could do something like this instead:

```ruby
def seed_files
  Vmdb::Plugins.flat_map.unshift(Rails) do |plugin|
    seed_files_for plugin.root
  end
end

def seed_files_for pathname
  Dir.glob(dir.join("product", "{report|compare}", "**", "*.yaml")).sort
end
```

Reply:
:+1:

Reply:
Only weird part might be the sorting, but I'm not sure that's important with respect to reports coming before compare or not.